doc/news/changes/minor/20200102BrunoTurcksin

```diff
@@ -0,0 +1,4 @@
+New: Template the function MatrixFree::get_dof_handler() to return either a
+DoFHandler<dim> or an hp::DoFHandler<dim>
```

Comment:
Maybe write:
```
New: The function MatrixFree::get_dof_handler() has been templated to enable
to return either a DoFHandler<dim> or an hp::DoFHandler<dim>.
```

Reply:
@Rombur I am curious in which context are you using this function? Either you are using `MatrixFree` as a container so that you do not have to pass the dof-handler explicitly as argument  to functions or you are accessing some functions for the dof-handler which are not provided by `MatrixFree`. In latter case, it might be an option - independent from this PR - to add that functionality...

Reply:
@kronbichler In a small side project, I am having a look into the hp-implementations. I'll keep you posted how user-friendly `MatrixFree` is for a matrix-free operator (once setup with a regular `DoFHandler` and once with `hp::DoFHandler`).

Reply:
@peterrum I am using it as a "container". I don't want to have to keep track of the `DoFHandler` everywhere.

Reply:
@Rombur If you add @kronbichler's suggestion, we can merge.